assets/src/components/stepperBar.tsx

```diff
@@ -0,0 +1,187 @@
+import useComponentSize from "@rehooks/component-size"
+import React, { useRef } from "react"
+import { SvgProps } from "../helpers/bsIcons"
+
+export const StepperBar = ({
+  totalSteps,
+  currentStep,
+}: {
+  totalSteps: number
+  currentStep: number
+}) => {
+  const stepperRef = useRef(null)
+  const { width } = useComponentSize(stepperRef)
+
+  const stepWidth = width / totalSteps
+
+  let stepper
+  if (currentStep == totalSteps) {
+    stepper = <FilledStepper stepWidth={stepWidth} stepCount={currentStep} />
+  } else {
+    stepper = (
+      <>
+        <FilledStepper
+          stepWidth={stepWidth}
+          stepCount={currentStep}
+          isPartial={true}
+        />
+        <UnfilledStepper
+          stepWidth={stepWidth}
+          stepCount={totalSteps - currentStep}
+        />
+      </>
+    )
+  }
```

Comment:
Nit (non-blocking): You can get rid of the `let` here by using a ternary:


```suggestion
  const stepper =
    currentStep == totalSteps ? (
      <FilledStepper stepWidth={stepWidth} stepCount={currentStep} />
    ) : (
      <>
        <FilledStepper
          stepWidth={stepWidth}
          stepCount={currentStep}
          isPartial={true}
        />
        <UnfilledStepper
          stepWidth={stepWidth}
          stepCount={totalSteps - currentStep}
        />
      </>
    )
```